sklearn/metrics/scorer.py

```diff
@@ -298,6 +312,8 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
     >>> grid = GridSearchCV(LinearSVC(), param_grid={'C': [1, 10]},
     ...                     scoring=ftwo_scorer)
     """
+    if name is None:
+        name = score_func.__name__
```

Comment:
Are we under the assumption that each `score_func` will have a `__name__`? (Slight reference to [this comment](https://github.com/scikit-learn/scikit-learn/pull/6697/files#r62779463))

Is there a test to ensure that every scoring function that exists in the scikit-learn codebase has a `__name__` attribute?